include/particle.tcc

```diff
@@ -213,37 +214,49 @@ bool mpm::Particle<Tdim, Tnphases>::map_mass_momentum_to_nodes(unsigned phase) {
 template <unsigned Tdim, unsigned Tnphases>
 void mpm::Particle<Tdim, Tnphases>::compute_strain(unsigned phase, double dt) {
   // Strain rate
-  Eigen::VectorXd strain_rate = cell_->compute_strain_rate(bmatrix_, phase);
-  // dstrain
-  Eigen::Matrix<double, 6, 1> dstrain;
-  dstrain.setZero();
+  Eigen::VectorXd cell_strain_rate =
+      cell_->compute_strain_rate(bmatrix_, phase);
+
   // Set dimension of strain rate
+  Eigen::Matrix<double, 6, 1> strain_rate;
+  strain_rate.setZero();
   switch (Tdim) {
     case (1): {
-      dstrain(0) = strain_rate(0);
+      strain_rate(0) = cell_strain_rate(0);
       break;
     }
     case (2): {
-      dstrain(0) = strain_rate(0);
-      dstrain(1) = strain_rate(1);
-      dstrain(3) = strain_rate(2);
+      strain_rate(0) = cell_strain_rate(0);
+      strain_rate(1) = cell_strain_rate(1);
+      strain_rate(3) = cell_strain_rate(2);
       break;
     }
     default: {
-      dstrain = strain_rate;
+      strain_rate = cell_strain_rate;
       break;
     }
   }
 
+  this->assign_strain_rate(phase, strain_rate);
```

Comment:
strain_rate_.col(phase) = strain_rate;

Reply:
I think we should use `assign_strain_rate` because the same is for both `stress` and `strain`. 
In addition, we can set unit tests easier when we can assign it.

Reply:
I'd remove it, because anything that has ownership of particle allows for assigning an arbitrary `strain_rate_`, which is not good.